apps/api/src/app/events/services/workflow.queue.service.ts

```diff
@@ -58,17 +71,82 @@ export class WorkflowQueueService {
     this.worker.on('failed', async (job, e) => {
       await this.jobRepository.updateStatus(job.data._organizationId, job.data._id, JobStatusEnum.FAILED);
       await this.jobRepository.setError(job.data._organizationId, job.data._id, e);
+
+      const lastWebhookFilterRetry =
+        job.attemptsMade === this.DEFAULT_ATTEMPTS && e.message.includes(EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER);
+
+      if (lastWebhookFilterRetry) {
+        await this.handleLastFailedWebhookFilter(job, e);
+      }
     });
+
     this.queueScheduler = new QueueScheduler('standard', this.bullConfig);
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async handleLastFailedWebhookFilter(job: any, e: Error) {
+    await this.createExecutionDetails.execute(
+      CreateExecutionDetailsCommand.create({
+        ...CreateExecutionDetailsCommand.getDetailsFromJob(job.data),
+        detail: DetailEnum.WEBHOOK_FILTER_FAILED,
+        source: ExecutionDetailsSourceEnum.WEBHOOK,
+        status: ExecutionDetailsStatusEnum.PENDING,
+        isTest: false,
+        isRetry: false,
+        raw: JSON.stringify({ message: JSON.parse(e.message).message }),
+      })
+    );
+
+    if (!job?.data?.step?.shouldStopOnFail) {
+      await this.queueNextJob.execute(
+        QueueNextJobCommand.create({
+          parentId: job?.data._id,
+          environmentId: job?.data._environmentId,
+          organizationId: job?.data._organizationId,
+          userId: job?.data._userId,
+        })
+      );
+    }
+  }
+
   public async addToQueue(id: string, data: JobEntity, delay?: number | undefined) {
     const options: JobsOptions = {
       removeOnComplete: true,
       removeOnFail: true,
       delay,
+      backoff: {
+        type: 'webhookFilterBackoff',
+      },
     };
 
     await this.queue.add(id, data, options);
   }
 }
+
+const getBackoffStrategies = (createExecutionDetails) => {
+  return {
+    webhookFilterBackoff: async (attemptsMade, err, job) => {
+      if (!shouldBackoff(err)) {
+        return -1;
+      }
+
+      await createExecutionDetails.execute(
+        CreateExecutionDetailsCommand.create({
+          ...CreateExecutionDetailsCommand.getDetailsFromJob(job.data),
+          detail: DetailEnum.WEBHOOK_FILTER_FAILED_RETRY,
+          source: ExecutionDetailsSourceEnum.WEBHOOK,
+          status: ExecutionDetailsStatusEnum.PENDING,
+          isTest: false,
+          isRetry: true,
+          raw: JSON.stringify({ message: JSON.parse(err.message).message, attempt: attemptsMade }),
+        })
+      );
+
+      return Math.round(Math.random() * Math.pow(2, attemptsMade) * 1000);
+    },
+  };
+};
+
+function shouldBackoff(err) {
```

Comment:
Maybe we could think of gather this function and `jobBackoffNeeded` (apps/api/src/app/events/usecases/run-job/run-job.usecase.ts) that are the same somewhere to have a single entry point of definition.
Something like a Backoff service that could be passed to different use cases.